packages/app-layout/src/vaadin-app-layout.js

```diff
@@ -453,6 +454,7 @@ class AppLayout extends ElementMixin(ThemableMixin(ControllerMixin(PolymerElemen
     this._navbarSizeObserver.observe(this.$.navbarBottom);
 
     window.addEventListener('close-overlay-drawer', this.__closeOverlayDrawerListener);
+    window.addEventListener('keydown', this.__onDrawerKeyDown);
```

Comment:
Ideally, we would remove this event listener when the component is disconnected from the DOM. However, given that we don't do that for other surrounding listeners, it's not a big deal.